### PR TITLE
chore: migrating App Controller/public to ES6

### PR DESCRIPTION
**Pull request by maze-runnar**
fix for #3618 
in this PR i am migrating app/controller/public.js to ES6 . 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/controllers/public.js  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4725161&bid=15775099)

File: app/controllers/public.js [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
 import moment from 'moment';
-import { computed } from '@ember/object';
+import { action, computed } from '@ember/object';
 
-export default Controller.extend({
-  smallLead: computed('session.currentRouteName', function() {
-    if (this.get('session.currentRouteName')) {
-      return this.get('session.currentRouteName') !== 'public.index';
+export default class extends Controller {
+  @computed('session.currentRouteName')
```
Comment (vishakhanihore) [inline app/controllers/public.js]: observe every property which you are using in the function

Comment (kushthedude) [inline app/controllers/public.js]: Observe all the properties which are getting used in the `computed` function.

File: app/controllers/public.js [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
-  displayEndDate: computed('model.startsAtDate', 'model.endsAtDate', function() {
+    return null;
+  }
+  @computed('model.startsAtDate', 'model.endsAtDate')
+  get displayEndDate() {
     return !moment(this.model.startsAtDate).isSame(this.model.endsAtDate, 'minute');
-  }),
-  displaySideMenu: computed('session.currentRouteName', function() {
+  }
+  @computed('session.currentRouteName')
```
Comment (kushthedude) [inline app/controllers/public.js]: Observe all the properties which are getting used in the computed function.

Comment (maze-runnar) [reply app/controllers/public.js]: @kushthedude please help me with observers . I am not getting any reference that how to implement them here in ES6 .
the below one that i tried with js.
```
displayEndDateChanges: observer('model.startsAtDate', 'model.endsAtDate', function() {
    console.log(this.model.startsAtDate + " " + this.model.endsAtDate)
  }),
```

File: app/controllers/public.js [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
+  get displaySideMenu() {
     if (this.session.currentRouteName) {
       return this.session.currentRouteName !== 'public.cfs.new-session' && this.session.currentRouteName !== 'public.cfs.new-speaker' && this.session.currentRouteName !== 'public.cfs.edit-speaker' && this.session.currentRouteName !== 'public.cfs.edit-session';
     }
-  }),
-  actions: {
-    toggleMenu() {
-      this.toggleProperty('isMenuOpen');
-    }
+    return null;
```
Comment (kushthedude) [inline app/controllers/public.js]: Why this line

File: app/controllers/public.js (hunk 2 above)
Comment (kushthedude) [reply app/controllers/public.js]: Didn't quite get you here ?

File: app/controllers/public.js (hunk 3 above)
Comment (maze-runnar) [reply app/controllers/public.js]: it was showing error that function``` displaySideMenu ``` should always return .

File: app/controllers/public.js (hunk 2 above)
Comment (maze-runnar) [reply app/controllers/public.js]: > Observe all the properties which are getting used in the computed function.

@kushthedude what is meaning of `observe all properties` .If it is related to observers , how to implement that here ?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/controllers/public.js  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4725161&bid=15775099)

File: app/controllers/public.js (hunk 3 above)
Comment (kushthedude) [reply app/controllers/public.js]: Can we remove the above `if` conditional?

Comment (maze-runnar) [reply app/controllers/public.js]: yes , i think we can remove

Comment (maze-runnar) [reply app/controllers/public.js]: > Can we remove the above `if` conditional?

checked after removing if -
Screenshot from 2019-12-25 10-15-46
Screenshot from 2019-12-25 10-16-16

Comment (kushthedude) [reply app/controllers/public.js]: If it is working then commit the changes :+1:

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/controllers/public.js  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=4725161&bid=15775099)

**Comment by kushthedude** [comment]
Severe are the consequences if its corresponding components and mixins are not migrated.

**Pull request closed without merging**